ddtrace/debugging/_redaction.py

```diff
@@ -0,0 +1,85 @@
+import typing as t
+
+from ddtrace.debugging._expressions import DDCompiler
+from ddtrace.debugging._expressions import DDExpression
+from ddtrace.internal.utils.cache import LFUCache
+from ddtrace.internal.utils.cache import cached
+from ddtrace.settings.dynamic_instrumentation import config
+from ddtrace.settings.dynamic_instrumentation import normalize_ident
+
+
+# The following identifier represent function argument/local variable/object
+# attribute names that should be redacted from the payload.
+REDACTED_IDENTIFIERS = (
+    frozenset(
+        {
+            "password",
+            "passwd",
+            "pwd",
+            "secret",
+            "key",
+            "auth",
+            "credentials",
+            "token",
+            "ipaddress",
+            "sessionid",
+            "remoteaddr",
+            "cookie",
+            "auth",
+            "sid",
+            "srf",
+            "session",
+            "symfony",
+            "srf",
+            "jwt",
+        }
+    )
+    | config.redacted_identifiers
+)
+
+
+REDACTED_PLACEHOLDER = r"{redacted}"
+
+
+def subidents(ident: str, min_len: int = 2) -> t.Set[str]:
+    n = len(ident)
+    return {ident[i:j] for i in range(n + 1 - min_len) for j in range(i + min_len, n + 1)}
+
+
+@cached()
+def redact(ident: str) -> bool:
+    return normalize_ident(ident) in REDACTED_IDENTIFIERS
+
+
+@cached()
+def redact_type(_type: str) -> bool:
+    _re = config.redacted_types_re
+    if _re is None:
+        return False
+    return _re.search(_type) is not None
+
+
+class DDRedactedCompiler(DDCompiler):
```

Comment:
love this.

We need to sync with the java impl. The java impl throw an exception if the condition is try to access redacted type/value. and also generate evaluation error to report that case.
While here, we just return the {redacted} placeholder and not report evaluation errors